src/riak_kv_ts_svc.erl

```diff
@@ -516,6 +516,8 @@ sub_tsqueryreq(_Mod, DDL = ?DDL{table = Table}, SQL, State) ->
         {error, {too_many_subqueries, NQuanta, MaxQueryQuanta}} ->
             {reply, make_max_query_quanta_resp(NQuanta, MaxQueryQuanta), State};
 
+        {error, AReason} when is_atom(AReason) -> %%<< i.e. overload, primary_not_available
+            {reply, make_rpberrresp(?E_SUBMIT, atom_to_list(AReason)), State};
```

Comment:
I see that other calls to `make_rpberrresp` seem to use `to_string/1` instead of `atom_to_list`. Maybe we should first test for `{error, Reason} when is_list(Reason) ->` as the first clause and convert anything else using `to_string/1` as the second clause.  That would catch any other reason, like a binary or record or whatever.